File: tests/machine/x/ts/dataset_sort_take_limit.ts
let products = [{name: "Laptop", price: 1500}, {name: "Smartphone", price: 900}, {name: "Tablet", price: 600}, {name: "Monitor", price: 300}, {name: "Keyboard", price: 100}, {name: "Mouse", price: 50}, {name: "Headphones", price: 200}];
let expensive = (() => {
  const _tmp1 = [];
  for (const p of products) {
    _tmp1.push({item: p, key: (-p.price)});
  }
  let res = _tmp1;
  res = res.sort((a,b)=> a.key < b.key ? -1 : a.key > b.key ? 1 : 0).map(x=>x.item);
  res = res.slice(1, (1 + 3));
  return res;
})()
;
console.log("--- Top products (excluding most expensive) ---");
const _tmp2 = expensive;
for (const item of (Array.isArray(_tmp2) ? _tmp2 : Object.keys(_tmp2))) {
  console.log(item.name, "costs $", item.price);
}

File: tests/machine/x/rust/dataset_sort_take_limit.rs
#[derive(Default, Debug, Clone, PartialEq)]
struct Product {
    name: &'static str,
    price: i32,
}

fn main() {
    let products = vec![Product { name: "Laptop", price: 1500 }, Product { name: "Smartphone", price: 900 }, Product { name: "Tablet", price: 600 }, Product { name: "Monitor", price: 300 }, Product { name: "Keyboard", price: 100 }, Product { name: "Mouse", price: 50 }, Product { name: "Headphones", price: 200 }];
    let expensive = { let mut tmp1 = Vec::new();for p in &products { let tmp2 = p.clone(); let tmp3 = -p.price; tmp1.push((tmp3, tmp2)); } tmp1.sort_by(|a,b| a.0.partial_cmp(&b.0).unwrap()); let mut tmp4 = Vec::new(); for p in tmp1 { tmp4.push(p.1); } let tmp4 = tmp4[1 as usize..(1 + 3) as usize].to_vec(); tmp4 };
    println!("{}", "--- Top products (excluding most expensive) ---");
    for item in expensive {
        println!("{:?} {} {:?}", item.name, "costs $", item.price);
    }
}
